### PR TITLE
Require xulieta and php

### PR DESCRIPTION
**Pull request by szepeviktor**
- we cannot run without xulieta
- we cannot run without php :)

### CODE REVIEW

**Comment by malukenho** [comment]
You should sign your commits (better with GPG) `git commit -s`

**Comment by szepeviktor** [comment]
I thought it is enough:

kép

**Comment by szepeviktor** [comment]
Could you turn it off for me - and maybe for everyone - it makes PR-s made with GutHub fail.

File: composer.json [hunk 1]
```diff
@@ -7,8 +7,11 @@
             "Codelicia\\XulietaJson\\": "src/"
         }
     },
-    "require-dev": {
-        "doctrine/coding-standard": "^8.2.0",
+    "require": {
+        "php": ">=7.4 || ~8.0",
         "codelicia/xulieta": "^0.1.3"
```
Comment (malukenho) [inline composer.json:12]: `codelicia/xulieta` should stay under `require-dev`.
For plugins, we just need it for development.

File: composer.json [hunk 2]
```diff
@@ -7,8 +7,11 @@
             "Codelicia\\XulietaJson\\": "src/"
         }
     },
-    "require-dev": {
-        "doctrine/coding-standard": "^8.2.0",
+    "require": {
+        "php": ">=7.4 || ~8.0",
```
Comment (malukenho) [inline composer.json:11]: May we go for PHP 8.1 only? /cc @EHER

**Review by malukenho — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
> Could you turn it off for me - and maybe for everyone? It makes PR-s made with GitHub fail.

That is actually easily done just by configuring `GIT_AUTOR` and `GIT_EMAIL`. I wouldn't go for getting rid of it, as it is a contract.

**Comment by malukenho** [comment]
If you have problems with that, I can sign it when merging.

File: composer.json (hunk 1 above)
Comment (szepeviktor) [reply composer.json:12]: So requiring `codelicia/xulieta-json` will not be enough? Users should always add the main package?

Comment (malukenho) [reply composer.json:12]: That is our idea. So people can plug in only what they need.

Comment (szepeviktor) [reply composer.json:12]: > what they need.

Is there such a use case when the main package is not needed?
I maintain several plugins and all of them depend on the main package.

Comment (malukenho) [reply composer.json:12]: Ah! I was talking about the plugins 😊

I can't think of a case where the main package is not required, but I want to minimize as much as possible the risk of conflict, so we can try different versions.

Comment (szepeviktor) [reply composer.json]: Reverted with sadness.

Comment (szepeviktor) [reply composer.json]: DCO: I'm sorry, I do it only here on GitHub's web interface...

**Review by malukenho — APPROVED** [review]

**Comment by malukenho** [comment]
Thank you @szepeviktor! Sorry for being sad about this.

**Comment by szepeviktor** [comment]
All right, no problem.
I think people seeing DCO will simply won't contribute...

**Comment by malukenho** [comment]
@all-contributors please add @szepeviktor for code.

**Pull request merged**